File: Anurag_19_CSE/Python/day28.py
# Link For Problem: https://leetcode.com/problems/missing-number/

class Solution:

    """
        TC : O(n)

        SC : O(1)
    """

    def missingNumber(self, nums: list[int]) -> int:
        n: int = len(nums)

        return (n*(n+1))//2 - sum(nums)

File: Anurag_19_CSE/Java/day28.java
// Link For Problem: https://leetcode.com/problems/missing-number/

public class day28 {

    /*
     * TC : O(n)
     * 
     * SC : O(1)
     */

    public int missingNumber(int[] nums) {
        int sum = 0;
        int n = nums.length;

        for (int i : nums)
            sum += i;

        return (n * (n + 1)) / 2 - sum;
    }
}
